Applications/DataExplorer/DataView/GEOModels.cpp

```diff
@@ -74,114 +81,87 @@ void GEOModels::updateGeometry(const std::string &geo_name)
-                                std::map<std::string,std::size_t>* ply_names )
+void GEOModels::addPolylineVec(const std::string& name)
 {
-	GEOObjects::addPolylineVec(lines, name, ply_names);
-	if (lines->empty())
-		return;
-
-	_geoModel->addPolylineList(QString::fromStdString(name), *this->getPolylineVecObj(name));
+	_geoModel->addPolylineList(QString::fromStdString(name),
+	                           *_geo_objects->getPolylineVecObj(name));
 	emit geoDataAdded(_geoModel, name, GeoLib::GEOTYPE::POLYLINE);
 }
 
-bool GEOModels::appendPolylineVec(const std::vector<GeoLib::Polyline*> &polylines,
-                                  const std::string &name)
+void GEOModels::appendPolylineVec(std::string const& name)
 {
-	bool ret (GeoLib::GEOObjects::appendPolylineVec (polylines, name));
-
-	this->_geoModel->appendPolylines(name, *this->getPolylineVecObj(name));
-	return ret;
+	this->_geoModel->appendPolylines(name,
+	                                 *_geo_objects->getPolylineVecObj(name));
 }
 
-bool GEOModels::removePolylineVec( const std::string &name )
+void GEOModels::removePolylineVec(std::string const& name)
 {
 	emit geoDataRemoved(_geoModel, name, GeoLib::GEOTYPE::POLYLINE);
 	this->_geoModel->removeGeoList(name, GeoLib::GEOTYPE::POLYLINE);
-	return GEOObjects::removePolylineVec (name);
 }
 
-void GEOModels::addSurfaceVec( std::vector<GeoLib::Surface*>* surfaces,
-                               const std::string &name,
-                               std::map<std::string,std::size_t>* sfc_names )
+void GEOModels::addSurfaceVec(std::string const& name)
 {
-	GEOObjects::addSurfaceVec(surfaces, name, sfc_names);
-	if (surfaces->empty())
-		return;
-
-	_geoModel->addSurfaceList(QString::fromStdString(name), *this->getSurfaceVecObj(name));
+	_geoModel->addSurfaceList(QString::fromStdString(name),
+	                          *_geo_objects->getSurfaceVecObj(name));
 	emit geoDataAdded(_geoModel, name, GeoLib::GEOTYPE::SURFACE);
 }
 
-bool GEOModels::appendSurfaceVec(const std::vector<GeoLib::Surface*> &surfaces,
-                                 const std::string &name)
+void GEOModels::appendSurfaceVec(std::string const& name)
 {
-	bool ret (GeoLib::GEOObjects::appendSurfaceVec (surfaces, name));
-
-	if (ret)
-		this->_geoModel->appendSurfaces(name, *this->getSurfaceVecObj(name));
-	else
+	_geoModel->appendSurfaces(name, *_geo_objects->getSurfaceVecObj(name));
+	/*
+	 * Before separation of GEOObjects from GEOModels if the
+	 * GEOObjects::appendSurfaceVec returned false the following code was
+	 * called:
 	{
-		std::vector<GeoLib::Surface*>* sfc = new std::vector<GeoLib::Surface*>;
-		for (std::size_t i = 0; i < surfaces.size(); i++)
-			sfc->push_back(surfaces[i]);
-		this->addSurfaceVec(sfc, name);
+	    std::vector<GeoLib::Surface*>* sfc = new std::vector<GeoLib::Surface*>;
+	    for (std::size_t i = 0; i < surfaces.size(); i++)
+	        sfc->push_back(surfaces[i]);
+	    _geo_objects->addSurfaceVec(sfc, name);
 	}
-
-	return ret;
+	*/
```

Comment:
Not sure what to do about this special handling. It is special because the extra `addSurfaceVec` calls were executed only when the code was compiled with `OGS_BUILD_GUI` flag set.